client/src/components/employer-component/box-row-component/index.jsx

```diff
@@ -0,0 +1,91 @@
+import { Grid, Typography, TextField } from "@mui/material";
+import ContentCopyIcon from "@mui/icons-material/ContentCopy";
+
+import PropTypes from "prop-types";
+import { BoxDivider } from "./index.styles";
+
+function BoxRowComponent({
+  leftSide,
+  rightSide,
+  copyable,
+  editable,
+  setRightSide,
+  rightSideWrapper,
+  required,
+}) {
+  return (
+    <>
+      <Grid
+        container
+        justifyContent="space-between"
+        alignItems="center"
+        sx={{ mb: 1, paddingX: "20px" }}
+      >
+        <Grid item xs={4} container alignItems="center">
+          <span>
+            <Typography variant="body1" align="left">
+              {leftSide}
+            </Typography>
+          </span>
+        </Grid>
+        <Grid item xs={8}>
+          {editable ? (
+            <TextField
+              variant="outlined"
+              fullWidth
+              value={rightSide}
+              onChange={(event) => setRightSide(event.target.value)}
+              disabled={!editable}
+              required={required}
+            />
+          ) : (
+            <div
+              style={{
+                display: "flex",
+                justifyContent: "space-between",
+              }}
+            >
+              <Typography variant="body1" align="left">
+                {rightSideWrapper(rightSide)}
+              </Typography>
+              {copyable && (
+                <ContentCopyIcon
+                  sx={{ color: "gray", cursor: "pointer" }}
+                  align="right"
+                  onClick={() => {
```

Comment:
It is currently crashing when the user wants to copy from clipboard I think the following will fix it 

```
 onClick={() => {
                    if (rightSide) {
                      navigator.clipboard.writeText(rightSide);
                    }
                  }}
```

Reply:
Good catch, thank you 😅 . I forgot to change this when I rewrote the component to use a wrapper for the right side 😆